src/utilities/oas-validator.ts

```diff
@@ -152,66 +224,84 @@ class OASValidator {
+      failures = [...failures, new InvalidSchema(ITEMS_MISSING_ERROR, path)];
+    }
+    return failures;
+  }
 
-    if (Array.isArray(actual)) {
-      // check that the expected object's items property is set
-      const itemSchema = expected.items;
-      if (itemSchema) {
-        // re-run for each item
-        actual.forEach((item) => {
-          failures = [
-            ...failures,
-            ...this.validateObjectAgainstSchema(item, itemSchema, path),
-          ];
-        });
-      } else {
-        failures.push(new Schema(ITEMS_MISSING_ERROR, path));
-      }
-    } else if (actualType === 'object') {
-      // check that the expected object's properties field is set
-      const properties = expected.properties;
-      if (properties) {
-        const actualProperties = Object.keys(actual);
-        const expectedProperties = Object.keys(properties);
-
-        // check that the actual object only contains properties present in expected object
-        if (
-          actualProperties.filter(
-            (property) => !expectedProperties.includes(property),
-          ).length > 0
-        ) {
-          failures.push(
-            new PropertiesMismatch(path, expectedProperties, actualProperties),
-          );
-        }
-
-        // check required values are present
-        expected.required?.forEach((requiredProperty) => {
-          if (!actualProperties.includes(requiredProperty)) {
-            failures.push(new RequiredProperty(path, requiredProperty));
-          }
-        });
-
-        // re-un for each property that has a schema present
-        Object.entries(actual)
-          .filter(([propertyName]) => properties[propertyName])
-          .forEach(([propertyName, propertyObject]) => {
-            path.push(propertyName);
-            failures = [
-              ...failures,
-              ...this.validateObjectAgainstSchema(
-                propertyObject,
-                properties[propertyName],
-                path,
-              ),
-            ];
-          });
-      } else {
-        failures.push(new Schema(PROPERTIES_MISSING_ERROR, path));
-      }
+  private static checkObjectPropertiesErrors(
+    actual: Json,
+    expected: SchemaObject,
+    path: string[],
+  ): ValidationFailure[] {
+    let failures: ValidationFailure[] = [];
+    const properties = expected.properties;
+
+    // check that the expected object's properties field is set
+    if (!properties) {
+      failures = [
+        ...failures,
+        new InvalidSchema(PROPERTIES_MISSING_ERROR, path),
+      ];
+      return failures;
+    }
+
+    const actualProperties = Object.keys(actual);
+    const expectedProperties = Object.keys(properties);
+
+    // check that the actual object only contains properties present in expected object
+    if (
+      actualProperties.filter(
+        (property) => !expectedProperties.includes(property),
+      ).length > 0
+    ) {
+      failures = [
+        ...failures,
```

Comment:
I don't think this is needed

Reply:
I put it there to be consistent in case we add another type of error in front of this one, but I can remove it if you consider it to be a waste.